flutter_local_notifications/lib/src/platform_flutter_local_notifications.dart

```diff
@@ -473,7 +473,7 @@ class AndroidFlutterLocalNotificationsPlugin
   /// Only [DrawableResourceAndroidIcon] and [ContentUriAndroidIcon] are
   /// supported for [AndroidIcon] fields.
   Future<MessagingStyleInformation?> getActiveNotificationMessagingStyle(
-    int id, {
+  {int? id,
```

Comment:
Can you elaborate on why this change was this done? You mentioned it can't be null to begin with so don't understand the reason behind this change that is separate to the main issue this PR is trying to resolve. Integer IDs are mandatory and tags are optional on Android as well so this is the reason why the method signature was originally the way it was. See https://developer.android.com/reference/androidx/core/app/NotificationManagerCompat#notify(java.lang.String,int,android.app.Notification) that documents the Android API to show a notification

Reply:
Since the `id` field in the `ActiveNotification` in passed to this function, it would take an `int` and receive an `int?` instead. 

The alternative is to once again use a default value/magic number to pass to this function, and explicitly note (in a comment/doc) that since the value is not nullable on Android and it's Android only, that value should never be used.

Which I think is a bit messy, so I opted to use a nullable id in the function signature instead.

Reply:
Whilst I get what you mean how this API may be used, by first calling `getActiveNotifications()`, if it's null then this would mean it's notification on iOS and therefore should be ignored i.e. `getActiveNotificationMessagingStyle()` shouldn't be called. This would equate to doing a null check so doesn't require the signature of `getActiveNotificationMessaginStyle()` to be changed. The reason why`id` in `ActiveNotification` became nullable to indicate the notification represents a iOS notification and/or that wasn't created by the plugin. If it's an Android notification, it always has an `id`.  By changing `id` to be nullable for `getActiveNotificationMessagingStyle()` that is  this implies that an Android notification can have a null `id`. This isn't true and can cause more confusion as it's an Android-specific API to begin with and isn't exposed by the cross-platform facing API. By that I mean you can't access the method via the [`FlutterLocalNotificationsPlugin` class] (https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/FlutterLocalNotificationsPlugin-class.html.) as you have to go via the [Android-specific implementation](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/AndroidFlutterLocalNotificationsPlugin-class.html). Hope that makes sense so that this particular change can be reverted

Reply:
Makes a lot of sense. I did add a comment to the example app, where I also added a null check `!` operator, so that it may clear things up for anyone trying to figure out what's going on.